### PR TITLE
Autonat v2 transport changes

### PR DESCRIPTION
**Pull request by umgefahren**
This is the PR for the transport redesign. I hope I *gited* it correctly.

This is a follow up to #1

### CODE REVIEW

File: misc/server/src/behaviour.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![allow(deprecated)]
+
```
Comment (thomaseizinger) [inline misc/server/src/behaviour.rs:2]: What are we deprecating here that can't be fixed in this PR? As a general rule, we should fix warnings in our own code. If we can't then our users won't be able to either.

File: swarm/src/lib.rs
Comment (thomaseizinger) [inline swarm/src/lib.rs:1]: Can we avoid these diffs?

**Review by thomaseizinger — APPROVED** [review]
LGTM

Two comments but we can also resolvs those in the upstream PR!

File: swarm/src/lib.rs
Comment (umgefahren) [reply swarm/src/lib.rs:1]: I think that's just formatting, but sure

File: misc/server/src/behaviour.rs (hunk 1 above)
Comment (umgefahren) [reply misc/server/src/behaviour.rs:2]: I think that's some relic from the AutoNATv2 changes

Comment (umgefahren) [reply misc/server/src/behaviour.rs:2]: I removed that

**Comment by umgefahren** [comment]
@thomaseizinger Can we merge that into the rust-libp2p PR?

**Comment by thomaseizinger** [comment]
> @thomaseizinger Can we merge that into the rust-libp2p PR?

Yes!

**Pull request merged**